src/nvim/api/options.c

```diff
@@ -309,15 +309,55 @@ Dictionary nvim_get_all_options_info(Error *err)
 ///     - commalist: List of comma separated values
 ///     - flaglist: List of single char flags
 ///
+/// The last set information applies to the local value in the current buffer
+/// or window if it is available, otherwise the global value information is
+/// returned.
 ///
 /// @param          name Option name
 /// @param[out] err Error details, if any
 /// @return         Option Information
 Dictionary nvim_get_option_info(String name, Error *err)
   FUNC_API_SINCE(7)
 {
-  return get_vimoption(name, err);
+  return get_vimoption(name, curbuf, curwin, err);
 }
+
+/// Gets the option information for one option from arbitrary buffer or window
+///
+/// This function is similar to @ref nvim_get_option_info except that it
+/// accepts an additional options table {opts}, allowing to specify option
+/// scope, buffer number or window used for retrieving the last set location
+/// information.
+///
+/// @param name      Option name
+/// @param opts      Optional parameters
+///                  - scope: One of "global" or "local". Analogous to
+///                  |:setglobal| and |:setlocal|, respectively.
+///                  - win: |window-ID|. Used for getting window local options.
+///                  - buf: Buffer number. Used for getting buffer local options.
+///                         Implies {scope} is "local".
+/// @param[out] err Error details, if any
+/// @return         Option Information
```

Comment:
Does it also return the value? Seems worthwhile to be able to deprecate all of:

```
nvim_get_option()                                                     
nvim_get_option_info()  
nvim_get_option_value()
nvim_win_get_option*
nvim_buf_get_option*
```

Reply:
I think that is a separate concern; it doesn't make sense to conflate the two (especially when `nvim_get_option_value` is still work in progress).

Reply:
`nvim_get_option`, `nvim_win_get_option` and `nvim_buf_get_option` already have a plan to be deprecated.

I think this should be separate from `nvim_get_option_value`, for the main reason I want the return types to be different for each.